templates/zerver/register.html

```diff
@@ -124,10 +124,9 @@
                     {% endif %}
             </div>
             <div class="help-box margin-top">
+                {{ _("Letters, numbers, and dashes only.") }}
                 {% if realms_have_subdomains %}
-                    {{ _("The address you'll use to sign in to your organization.") }}
```

Comment:
I think we can keep this if-else structure. a-z, 0-9 etc in the non-subdomains case, and I think we don't have to say it in the subdomains case. 

"the address you'll use to sign in to your organization" does feel more clear than than the shorter version.